### ISSUE TITLE
[attributes-order] ngClass is being treated as STRUCTURAL_DIRECTIVE

### ISSUE THREAD

**Issue by hjbolide** [angular-eslint/angular-eslint#1914]
Hi, 

Background: The angular project I'm currently working on is still on v15, but because of this fix: [1448](https://github.com/angular-eslint/angular-eslint/pull/1448) is only available from v16 of eslint-plugin-template, I manually bumped the eslint related packages to v16.3.1 while the main project is still on v15, this allows me to execute `ng lint --fix` and get correct result in templates.

Question: I'm trying to use eslint to enforce and fix our angular templates, one rule I applied:

```JSON
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "order": [
          "STRUCTURAL_DIRECTIVE",
          "TEMPLATE_REFERENCE",
          "ATTRIBUTE_BINDING",
          "INPUT_BINDING",
          "TWO_WAY_BINDING",
          "OUTPUT_BINDING"
        ]
      }
    ]
  }
}
```

And what we want to achieve is this:

Original template snippet:
```HTML
<span *ngIf="index > 0"
      [ngClass]="{validCopy: day.valid, invalidCopy: !day.valid}"
      (click)="copyTime(day.valid, false, 'left', index)"
      title="{{'Copy to all previous days' | messageTranslate}}">
  <fa-icon icon="angle-double-left"/>
</span>
```

Expected template snippet after `ng lint --fix`:
```HTML
<span *ngIf="index > 0"
      title="{{'Copy to all previous days' | messageTranslate}}"
      [ngClass]="{validCopy: day.valid, invalidCopy: !day.valid}"
      (click)="copyTime(day.valid, false, 'left', index)">
  <fa-icon icon="angle-double-left"/>
</span>
```

Actual template snippet after `ng lint --fix`:
```HTML
<span *ngIf="index > 0"
      [ngClass]="{validCopy: day.valid, invalidCopy: !day.valid}"
      title="{{'Copy to all previous days' | messageTranslate}}"
      (click)="copyTime(day.valid, false, 'left', index)">
  <fa-icon icon="angle-double-left"/>
</span>
```

It just seemed like the `[ngClass]` had been categorised as structural directive instead of input binding.

Wondering if anyone has any insight on this?

Thank you very much,
Charles

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `16.3.1` |
| `@angular-eslint/template-parser`         | `16.3.1` |
| `@typescript-eslint/parser`               | `5.62.0` |
| `ESLint`                                  | `8.57.0` |
| `node`                                    | `18.16.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
@angular-devkit/architect       0.1502.10
@angular-devkit/build-angular   15.2.10
@angular-devkit/core            15.2.10
@angular-devkit/schematics      15.2.10
@angular/cdk                    15.2.9
@schematics/angular             15.2.10
rxjs                            7.8.1
typescript                      4.9.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by reduckted** [comment]
Interesting. The problem is actually that `title` is being treated as an input binding. You can see here:
https://astexplorer.net/#/gist/6f1cd5d0f7e4e03bfffb71a54c1e540f/1bfa281ff1ef7b725ca28bbe6b4fe257256d03ba

I assume it's an input binding because it's value is calculated rather than static (because of the `{{ ... }}`). Maybe inputs need an additional check that the attribute name is in square brackets? If it's not, then treat it as an attribute binding?

**Comment by hjbolide** [comment]
Hi @reduckted 

Thanks for pointing that out, and yes I think though semantically and technically "title" is, in this case an input binding, but given we are trying to rearrange the tag attributes, it should instead looking at how the attribute is written.

**Comment by reduckted** [comment]
Absolutely, and if #2307 is merged, then this change will _need_ to be made because if there are i18n attributes, then inputs are classified based on the square brackets and not what the Angular template parser decides.

It should be a simple change once #2307 is merged.

**Comment by hjbolide** [comment]
Thanks @reduckted, really appreciate it.
